libmcount/pyhook.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Python binding for mcount entry and exit
+ *
+ * Copyright (C) 2017, LG Electronics, Honggyu Kim <hong.gyu.kim@lge.com>
+ *
+ * Released under the GPL v2.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <dlfcn.h>
+
+#include "pyhook.h"
+#include "utils/utils.h"
+
+/* python file name */
+const char *pyfile;
+
+static bool python_enabled = false;
+
+static void *python_handle;
+static const char *libpython = "/usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so";
+
+static PyAPI_FUNC(void) (*__Py_Initialize)(void);
+static PyAPI_FUNC(void) (*__PySys_SetPath)(char *);
+static PyAPI_FUNC(void) (*__PyErr_Print)(void);
+static PyAPI_FUNC(int) (*__PyCallable_Check)(PyObject *);
+static PyAPI_FUNC(PyObject *) (*__PyImport_Import)(PyObject *name);
+static PyAPI_FUNC(PyObject *) (*__PyTuple_New)(Py_ssize_t size);
+static PyAPI_FUNC(PyObject *) (*__PyErr_Occurred)(void);
+static PyAPI_FUNC(int) (*__PyTuple_SetItem)(PyObject *, Py_ssize_t, PyObject *);
+static PyAPI_FUNC(PyObject *) (*__PyObject_GetAttrString)(PyObject *, const char *);
+static PyAPI_FUNC(PyObject *) (*__PyObject_CallObject)(PyObject *callable_object, PyObject *args);
+
+static PyAPI_FUNC(PyObject *) (*__PyString_FromString)(const char *);
+static PyAPI_FUNC(PyObject *) (*__PyInt_FromLong)(long);
+
+static PyAPI_FUNC(char *) (*__PyString_AsString)(PyObject *);
+static PyAPI_FUNC(long) (*__PyLong_AsLong)(PyObject *);
+
+static PyObject *pName, *pModule, *pFuncEntry, *pFuncExit;
+
+#define INIT_PY_API_FUNC(func) \
+	do { \
+		__##func = dlsym(python_handle, #func); \
+		if (!__##func) { \
+			pr_err("dlsym for " #func " is failed!\n"); \
+			python_enabled = false; \
+			return -1; \
+		} \
+	} while (0)
+
+int python_init(void)
+{
+	if (!pyfile) {
+		python_enabled = false;
+		return -1;
+	}
+
+	python_handle = dlopen(libpython, RTLD_LAZY);
+	if (!python_handle) {
+		pr_warn("libpython.so cannot be loaded!\n");
+		return -1;
+	}
+
+	INIT_PY_API_FUNC(Py_Initialize);
+	INIT_PY_API_FUNC(PySys_SetPath);
+	INIT_PY_API_FUNC(PyErr_Print);
+	INIT_PY_API_FUNC(PyCallable_Check);
+	INIT_PY_API_FUNC(PyImport_Import);
+	INIT_PY_API_FUNC(PyTuple_New);
+	INIT_PY_API_FUNC(PyErr_Occurred);
+
+	INIT_PY_API_FUNC(PyTuple_SetItem);
+	INIT_PY_API_FUNC(PyObject_GetAttrString);
+	INIT_PY_API_FUNC(PyObject_CallObject);
+
+	INIT_PY_API_FUNC(PyString_FromString);
+	INIT_PY_API_FUNC(PyInt_FromLong);
+	INIT_PY_API_FUNC(PyString_AsString);
+	INIT_PY_API_FUNC(PyLong_AsLong);
+
+	__Py_Initialize();
+
+	//PySys_SetPath("/usr/local/modules");
+	__PySys_SetPath("./");  // path to the module to import
```

Comment:
I got the same output.  In addition, entering to the tests directory and running the python script didn't show any output:

```
$ cd tests
$ uftrace -p mcount_replay -R ".*@retval" t-abc
# DURATION    TID     FUNCTION
            [22021] | main() {
            [22021] |   a() {
            [22021] |     b() {
            [22021] |       c() {
   2.507 us [22021] |         getpid() = 22021;
  21.885 us [22021] |       } /* c */
  22.392 us [22021] |     } /* b */
  22.781 us [22021] |   } /* a */
  24.193 us [22021] | } /* main */
```